src/sparseml/pytorch/sparsification/distillation/modifier_per_layer.py

```diff
@@ -0,0 +1,368 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Modifier for performing knowledge distillation via feature imitation.
+"""
+
+import logging
+from typing import Any, List, Optional, Union, Mapping
+
+import torch
+from torch.nn import Module
+
+from sparseml.optim import ModifierProp
+from sparseml.pytorch.sparsification.distillation.modifier_distillation_base import (
+    BaseDistillationModifier,
+)
+from sparseml.pytorch.sparsification.modifier import PyTorchModifierYAML
+from sparseml.pytorch.utils import BaseLogger
+
+
+__all__ = [
+    "PerLayerDistillationModifier",
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+_DISTILLATION_TYPES = [
+    torch.nn.Conv1d,
+    torch.nn.Conv2d,
+    torch.nn.Conv3d,
```

Comment:
we only actually support Conv2d i believe (the other two don't have 4d input sizes so the projections won't work).